SFR-1675: Playwright tests on homepage collections

clarissarichard opened the pull request:
[Jira Ticket](http://jira.nypl.org/browse/SFR-1675)

### This PR does the following:
- Adds Playwright tests to check the collections section on the homepage

### Testing requirements & instructions: 
-  Running Playwright tests and all tests pass

samanthaandrews commented on playwright/tests/homePageElements.spec.ts:
```diff
@@ -73,6 +73,47 @@ Then("I see advanced search link", async function (this: CustomWorld) {
   await expect(this.page.locator("text=Advanced Search")).toBeVisible();
 });
 
+// 1. Write assertion to find Recently Added Collections heading
+// 2. Confirm heading is visible
```
Now that the test is written and passing, let's remove these todo list comments

samanthaandrews commented on playwright/tests/homePageElements.spec.ts:
```diff
@@ -73,6 +73,47 @@ Then("I see advanced search link", async function (this: CustomWorld) {
+  await expect(this.page.getByRole('heading', { name: 'Recently Added Collections' })).toBeVisible();
+})
+
+// 1. Find a locator that is not dependent on user content
+// 2. Assert that locator is visible
+// Things we could assert: 
+// - there's at least one card
+// - the card has the text "Collection"
+// - the card is a link
+// - the card has the text "Items"
+// - the card has an image
```
Let's remove these comments as well

samanthaandrews commented on playwright/tests/homePageElements.spec.ts:
```diff
@@ -73,6 +73,47 @@ Then("I see advanced search link", async function (this: CustomWorld) {
+
+  // confirm correct image is located
+  const imgLocator = await this.page.getByRole("img", { name: "Collection" }).first().getAttribute("src");
+  console.log("collection card img src attribute:", imgLocator);
```
remove stray console.log

samanthaandrews commented on playwright/tests/homePageElements.spec.ts:
```diff
@@ -73,6 +73,47 @@ Then("I see advanced search link", async function (this: CustomWorld) {
+
+  // confirm correct "Items" text is located
+  const itemsLocator = await this.page.getByRole("heading", { name: /Items/ }).first().innerText();
+  console.log("collection card items text:", itemsLocator);
```
remove stray console.log

samanthaandrews reviewed (changes requested):
Just a couple of small cleanup things 👍 This looks really good!

clarissarichard commented:
@samanthaandrews thanks for reviewing my PR! I removed the console.logs and todo comments, then pushed my changes. From the failing checks, I have a few questions to better understand what to correct:
- Even though we've already included a bullet point describing the new collections tests, do I also need to note that I removed the comments in the change log or should that check be ignored in this case?
- It looks like lint is recommending to change single quotes to double quotes. Which of the two are preferred here?
- I believe the Test & Coverage check found an error with an Advanced Search test. If this happens on a file that I did not make changes to, how should I proceed?

samanthaandrews commented:
> Even though we've already included a bullet point describing the new collections tests, do I also need to note that I removed the comments in the change log or should that check be ignored in this case?

Nope, you just need to update the Changelog once. It should include the changes that are part of the whole PR that will be merged to the main branch. I don't see any changes to the Changelog in this branch though. Not sure what happened to it, but you'll want to add a line there and push it up again in order to get that check to pass.

> It looks like lint is recommending to change single quotes to double quotes. Which of the two are preferred here?

Yeah I think our linter is configured to prefer double quotes. Within VS Code you can install an extension called `eslint` that will call out these errors for you locally so you can fix them before pushing up to this branch again. Lmk if you need help getting this to work!

> I believe the Test & Coverage check found an error with an Advanced Search test. If this happens on a file that I did not make changes to, how should I proceed?

This test failure is expected because it also exists on my base branch. Don't worry about it!

Great work.
